mm2src/coins/lightning/ln_events.rs

```diff
@@ -461,21 +492,91 @@ impl LightningEventHandler {
                 },
             };
             let claiming_txid = spending_tx.txid().to_string();
-            let persister = self.persister.clone();
-            spawn(async move {
-                ok_or_retry_after_sleep!(
-                    persister
-                        .add_claiming_tx_to_db(
-                            closing_txid.clone(),
-                            claiming_txid.clone(),
-                            (claimed_balance as f64) - claiming_tx_fee_per_channel,
-                        )
-                        .await,
-                    TRY_LOOP_INTERVAL
-                );
-            });
+            let db = self.db.clone();
+            // This doesn't need to be respawned on restart unlike add_closing_tx_to_db since Event::SpendableOutputs will be re-fired on restart
+            // if the spending_tx is not broadcasted.
+            let abort_handler = spawn_abortable(add_claiming_tx_to_db_loop(
```

Comment:
Yeah, I guess so. I think I should use a different function other than `broadcast_transaction` of the `BroadcasterInterface` since this function doesn't return an error. Also, after implementing P2P transaction broadcasting, if regular broadcasting of the transaction fails and it got broadcasted through the P2P network, I guess I will have to check if the transaction is on chain or not before saving it to the DB.

Reply:
Done